woocommerce/woocommerce-ios
# HACK week: Add coupon to Order [1/2]

Part 1/2 of HACK week project: Add coupon to Order

## Description

This PR adds networking and Yosemite related changes for "Add coupon to Order" feature.

Changes
- New feature flag `addCouponToOrder`
- Make `OrderCouponLine` codable. (Only coupon code is a writable field)
- `OrderFactory` - Helper method to create coupon.
- Add `couponLines` to create and update order requests.

## Testing instructions

- This PR encodes `couponLines` and adds to the create and update order requests.
- For now, `couponLines` will not have any values, and we will use this in this PR #9090
- Please make sure that order creation and updating flows work like before.

## Screenshots
NA

---
